File: beginner/__1000__/1047/1047.lua
local startHour = io.read("*n")
local startMinute = io.read("*n")
local endHour = io.read("*n")
local endMinute = io.read("*n")

local startTime = startHour * 60 + startMinute
local endTime = endHour * 60 + endMinute

local diff

if startTime == endTime then
    diff = 24 * 60
elseif endTime > startTime then
    diff = endTime - startTime
else
    diff = (24 * 60 - startTime) + endTime
end

local hours = diff // 60
local minutes = diff % 60

print(string.format(
    "O JOGO DUROU %d HORA(S) E %d MINUTO(S)",
    hours,
    minutes
))

File: beginner/__1000__/1047/1047.cpp
#include <iostream>

using namespace std;

int main(){

    int h1, m1, h2, m2;
    cin >> h1 >> m1 >> h2 >> m2;

    int hr, min;

    if(h2 < h1 || (h2 == h1 && m2 < m1)){
        h2 += 24;
    }

    if(m2 < m1){
        m2 += 60;
        h2 -= 1;
    }

    hr = h2 - h1;
    min = m2 - m1;

    if(hr == 0 && min == 0){
        hr = 24;
        min = 0;
    }

    cout << "O JOGO DUROU "  << hr << " HORA(S) E " << min << " MINUTO(S)\n";

    return 0;
}